packages/core/aqua-to-js/src/generate/interfaces.ts

```diff
@@ -117,12 +119,19 @@ export class TSTypeGenerator implements TypeGenerator {
     const serviceDecl = `service: ${srvName}Def`;
     const serviceIdDecl = `serviceId: string`;
 
-    const registerServiceArgs = [
-      [serviceDecl],
-      [serviceIdDecl, serviceDecl],
-      [peerDecl, serviceDecl],
-      [peerDecl, serviceIdDecl, serviceDecl],
-    ];
+    const registerServiceArgs =
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+      (srvDef.defaultServiceId as DefaultServiceId).s_Some__f_value != null
+        ? [
+            [serviceDecl],
+            [serviceIdDecl, serviceDecl],
+            [peerDecl, serviceDecl],
+            [peerDecl, serviceIdDecl, serviceDecl],
+          ]
+        : [
+            [serviceIdDecl, serviceDecl],
+            [peerDecl, serviceIdDecl, serviceDecl],
+          ];
```

Comment:
What does this mean? I suggest to:
1. Move condition to a separate named variable, explain what it means in comments
2. Add a type to `registerServiceArgs`, so it's clear from the type what you intend to create here
3. Explain different structures on different conditions in a comment

Right now it looks like black magic. Here you repeat `serviceDecl` two times, there three times, add a frog leg and Hippogriff feather, and violà!

Reply:
Moved these matrixes to a named variable with the descriptive naming.